app/plugin.go

```diff
@@ -489,59 +495,90 @@ func (a *App) getRemotePlugins(filter *model.MarketplacePluginFilter) ([]*model.
 			continue
 		}
 
-		result = append(result, &model.MarketplacePlugin{BaseMarketplacePlugin: p})
+		result[p.Manifest.Id] = &model.MarketplacePlugin{BaseMarketplacePlugin: p}
 	}
 
 	return result, nil
 }
 
-// addLocalPlugins returns a merged list of locally installed and remote marketplace plugins.
-func (a *App) addLocalPlugins(remoteMarketplacePlugins []*model.MarketplacePlugin) ([]*model.MarketplacePlugin, *model.AppError) {
-	allPlugins := map[string]*model.MarketplacePlugin{}
+// mergePrepackagePlugins returns a merged list of pre-packaged plugins and remote marketplace plugins.
+func (a *App) mergePrepackagePlugins(remoteMarketplacePlugins map[string]*model.MarketplacePlugin) *model.AppError {
+	pluginsEnvironment := a.GetPluginsEnvironment()
+	if pluginsEnvironment == nil {
+		return model.NewAppError("GetMarketplacePlugins", "app.plugin.config.app_error", nil, "", http.StatusInternalServerError)
+	}
 
-	for _, p := range remoteMarketplacePlugins {
-		if p.Manifest == nil {
+	for _, prepackaged := range pluginsEnvironment.PrepackagedPlugins() {
+		if prepackaged.Manifest == nil {
+			continue
+		}
+
+		prepackagedMarketplace := &model.MarketplacePlugin{
+			BaseMarketplacePlugin: &model.BaseMarketplacePlugin{
+				Manifest: prepackaged.Manifest,
+			},
+		}
+
+		// If not available in marketplace, add the prepackaged
+		if remoteMarketplacePlugins[prepackaged.Manifest.Id] == nil {
+			remoteMarketplacePlugins[prepackaged.Manifest.Id] = prepackagedMarketplace
 			continue
 		}
-		allPlugins[p.Manifest.Id] = p
+
+		// If available in the markteplace, only overwrite if newer.
+		var prepackagedVersion, marketplaceVersion semver.Version
```

Comment:
is it required to pre-define these?

Reply:
Not really, it's just that the short variable declaration is tricky when I'm reusing the `err` below. Someone needs to be predefined, either `marketplaceVersion` or `err` 🤔

Reply:
hmm does it? 🤔 maybe i'm missing something but i thought if you remove this and just switch `=` to `:=` in `marketplaceVersion, err = semver.Parse(marketplacePlugin.Manifest.Version)`, that should do it, no? 
i.e. `marketplaceVersion, err := semver.Parse(marketplacePlugin.Manifest.Version)`

Reply:
Right but for the `prepackagedVersion` a couple of lines above is also using the same err. See updated commit

Reply:
but i think that should be ok, as long as there is one new variable golang will let you use `:=`

Reply:
Got it! It kept giving me the error because I was not saving the file 🤦‍♀️

Reply:
heh, you should just use `"files.autosave": "onFocusChange"` setting in vscode. Saves automatically for you ;-)

![image](https://user-images.githubusercontent.com/25732808/71269633-cc7d2a00-231d-11ea-8918-272df779c175.png)